homeassistant/bootstrap.py

```diff
@@ -304,6 +305,20 @@ def _get_domains(hass: core.HomeAssistant, config: Dict[str, Any]) -> Set[str]:
 async def _async_set_up_integrations(
         hass: core.HomeAssistant, config: Dict[str, Any]) -> None:
     """Set up all the integrations."""
+    # Start up debuggers. Start these first in case they want to wait.
+    debuggers = [debug
+                 for debug in DEBUGGER_INTEGRATIONS
+                 if debug in config]
+    if debuggers:
+        _LOGGER.debug("starting up debuggers %s", debuggers)
+        await asyncio.gather(*[
+            loader.async_component_dependencies(hass, domain)
```

Comment:
Why this?

Reply:
Otherwise it fails and throws an exception. I didn't get too far into figuring out why, something to do with non aio wait in the aio thread. It also failed if you use setup instead of async_setup.

Reply:
So wait works and you can debug component setup if you want to. Otherwise by the time you connect it's already happened.

Reply:
Note that if there's not a wait in the debugger config, it just starts up quickly and returns, and if there's no debugger configured then nothing happens.

One could optimise it a bit by only starting up components here that have `wait=True` explicitly set in the config, but probs best just get on with it really.

Reply:
but why do you need to process the dependencies ? That's also done inside `async_setup_component` ?

Reply:
Oh OK fair enough then.